src/Classes/ServiceAPI/MyRadio_Timeslot.php

```diff
@@ -435,6 +435,38 @@ public static function getCurrentTimeslot($time = null, $filter = array(1))
         }
     }
 
+    /**
+     * Gets the previous Timeslots before $time, in reverse chronological order.
+     *
+     * @param int $time
+     * @param int $n defines the number of timeslots you want before this time.
+     * @param     $filter defines a filter of show_type ids
+     *
+     * @return MyRadio_Timeslot
+     */
+    public static function getPreviousTimeslots($time = null, $n = 1, $filter = array(1))
+    {
+        $filter = '{'.implode(', ', $filter).'}'; // lolphp http://php.net/manual/en/function.pg-query-params.php#71912
```

Comment:
I think this is in a few places in the codebase. Maybe make a static helper function in the database class?